src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -384,16 +386,100 @@
+
+(rf/reg-event-fx
+ :wallet.send/auth-slider-completed
+ (fn [{:keys [db]}]
    (let [last-request-uuid (get-in db [:wallet :ui :send :last-request-uuid])]
-     {:db (-> db
-              (assoc-in [:wallet :ui :send :amount] amount)
-              (update-in [:wallet :ui :send] dissoc :transaction-for-signing))
+     {:db (update-in db [:wallet :ui :send] dissoc :transaction-for-signing)
       :fx [[:dispatch [:wallet/build-transactions-from-route {:request-uuid last-request-uuid}]]
            [:dispatch
-            [:wallet/wizard-navigate-forward
-             {:current-screen stack-id
-              :start-flow?    start-flow?
-              :flow-id        :wallet-send-flow}]]]})))
+            [:wallet.send/set-sign-transactions-callback-fx
+             [:dispatch
+              [:wallet/prepare-signatures-for-transactions :send]]]]]})))
+
+(defn log-transaction-signature-error
+  [error]
+  (log/error
+   "failed to prepare signatures for transactions"
+   {:event :wallet/prepare-signatures-for-transactions
+    :error error})
+  (rf/dispatch
+   [:toasts/upsert
+    {:id   :prepare-signatures-for-transactions-error
+     :type :negative
+     :text (:message error)}]))
+
+(defn- send-transactions-with-signatures
+  [tx-type signatures]
+  (rf/dispatch
+   [:wallet/send-router-transactions-with-signatures tx-type
+    signatures]))
+
+(rf/reg-event-fx
+ :wallet/send-router-transactions-with-signatures
+ (fn [{:keys [db]} [tx-type signatures]]
+   (let [transaction-for-signing (get-in db [:wallet :ui tx-type :transaction-for-signing])
+         signatures-map          (reduce (fn [acc {:keys [message signature]}]
+                                           (assoc acc
+                                                  message
+                                                  (send-utils/signature-rsv signature)))
+                                         {}
+                                         signatures)]
+     {:json-rpc/call [{:method     "wallet_sendRouterTransactionsWithSignatures"
+                       :params     [{:uuid       (get-in transaction-for-signing [:sendDetails :uuid])
+                                     :signatures signatures-map}]
+                       :on-success (fn []
+                                     (rf/dispatch [:hide-bottom-sheet]))
+                       :on-error   (fn [error]
+                                     (log/error "failed to send router transactions with signatures"
+                                                {:event :wallet/send-router-transactions-with-signatures
+                                                 :error error})
+                                     (rf/dispatch [:toasts/upsert
+                                                   {:id   :send-router-transactions-with-signatures-error
+                                                    :type :negative
+                                                    :text (:message error)}]))}]})))
+
+(rf/reg-event-fx
```

Comment:
This is the key refactoring.
Before if user doesnt need keycard sign, we initiated standard authorization from slider component and than were signing from this event. But when user required keycard sign, we initiated keycard authorization from this event with consecutive sign.
Now both types of authorization run from here, that simplified caller code.